spring-xd-module/src/main/java/org/springframework/xd/module/spark/streaming/SparkStreamingDriverModule.java

```diff
@@ -106,51 +113,56 @@ public SparkStreamingDriverModule(ModuleDescriptor descriptor, ModuleDeploymentP
 		logger.info("Starting SparkDriver");
-		Environment env = this.getApplicationContext().getEnvironment();
-		final Receiver receiver = getComponent(Receiver.class);
-		final Processor processor = getComponent(Processor.class);
-		Properties sparkConfigs = getSparkModuleProperties(processor);
-		SparkConf sparkConf = new SparkConf()
-				// Set spark UI port to random available port to support multiple spark modules on the same host.
-				.set("spark.ui.port", String.valueOf(SocketUtils.findAvailableTcpPort()))
-				// Set the cores max so that multiple (at least a few) spark modules can be deployed on the same host.
-				.set("spark.cores.max", "3")
-				.setMaster(env.getProperty(Processor.SPARK_MASTER_URL_PROP, Processor.SPARK_DEFAULT_MASTER_URL))
-				.setAppName(getDescriptor().getGroup() + "-" + getDescriptor().getModuleLabel());
-		if (sparkConfigs != null) {
-			for (String property : sparkConfigs.stringPropertyNames()) {
-				sparkConf.set(property, sparkConfigs.getProperty(property));
-			}
-		}
-		List<String> sparkJars = new ArrayList<String>();
-		// Add jars from spark.jars (if any) set from spark module.
+		SparkStreamingSupport processor;
+		Properties sparkConfigs = null;
 		try {
-			String jarsFromConf = sparkConf.get("spark.jars");
-			if (StringUtils.hasText(jarsFromConf)) {
-				sparkJars.addAll(Arrays.asList(jarsFromConf.split("\\s*,\\s*")));
+			processor = getComponent(SparkStreamingSupport.class);
+			if (processor != null) {
+				sparkConfigs = getSparkModuleProperties(processor);
 			}
 		}
-		catch (NoSuchElementException e) {
-			// should ignore
+		catch (NoSuchBeanDefinitionException e) {
+			throw new RuntimeException("Either java or scala module should be present.");
 		}
-		sparkJars.addAll(getApplicationJars());
-		sparkConf.setJars(sparkJars.toArray(new String[sparkJars.size()]));
-		String batchInterval = env.getProperty("batchInterval",
-				env.getProperty("spark.streaming.batchInterval", SPARK_STREAMING_BATCH_INTERVAL));
-		this.streamingContext = new JavaStreamingContext(sparkConf, new Duration(Long.valueOf(batchInterval)));
+		startSparkStreamingContext(sparkConfigs, processor);
+	}
 
+	/**
+	 * Start spark streaming context for the given streaming processor.
+	 *
+	 * @param sparkConfigs the spark configuration properties
+	 * @param sparkStreamingSupport the underlying processor implementation
+	 */
+	private void startSparkStreamingContext(Properties sparkConfigs,
+			final SparkStreamingSupport sparkStreamingSupport) {
+		final Receiver receiver = getComponent(Receiver.class);
+		Environment env = this.getApplicationContext().getEnvironment();
+		String masterURL = env.getProperty(SparkStreamingSupport.SPARK_MASTER_URL_PROP,
+				SparkStreamingSupport.SPARK_DEFAULT_MASTER_URL);
+		final SparkConf sparkConf = setupSparkConf(masterURL, sparkConfigs);
+		final String batchInterval = env.getProperty("batchInterval",
+				env.getProperty("spark.streaming.batchInterval", SPARK_STREAMING_BATCH_INTERVAL));
```

Comment:
should we a constant (to SparkStreamingSupport) for the key: "spark.streaming.batchInterval"?